### PR TITLE
Upstream merge 2

### PR DESCRIPTION
**Pull request by j-mie**
Some upstream changes, tested and it works.

### CODE REVIEW

File: code/_compile_options.dm [hunk 1]
```diff
@@ -2,6 +2,11 @@
 								//Enables in-depth debug messages to runtime log (used for debugging)
 //#define TESTING				//By using the testing("message") proc you can create debug-feedback for people with this
 								//uncommented, but not visible in the release version)
+#ifdef TESTING
+//#define GC_FAILURE_HARD_LOOKUP	//makes paths that fail to GC call find_references before del'ing.
```
Comment (ghost) [inline code/_compile_options.dm:6]: why is this commented?

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/world.dm [hunk 2]
```diff
@@ -243,30 +248,40 @@ var/last_irc_status = 0
...
+	world << "<span class='boldannounce'>Rebooting world.</span>"
 	for(var/thing in clients)
 		var/client/C = thing
-		if (!C)
-			continue
-		if(config.server)	//if you set a server location in config.txt, it sends you there instead of trying to reconnect to the same world address. -- NeoFite
+		if(C && config.server)	//if you set a server location in config.txt, it sends you there instead of trying to reconnect to the same world address. -- NeoFite
 			C << link("byond://[config.server]")
 
+/world/proc/RoundEndSound(round_end_sound_sent)
```
Comment (ghost) [inline code/world.dm]: smh this casing looks autistic `why_not_make_it_like_this_like_most_procs_are`

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/_compile_options.dm (hunk 1 above)
Comment (j-mie) [reply code/_compile_options.dm:6]: Check: https://github.com/tgstation/tgstation/blob/master/code/_compile_options.dm#L6-L10

File: code/world.dm [hunk 3]
```diff
@@ -243,30 +246,42 @@ var/last_irc_status = 0
...
 		var/client/C = thing
-		if (!C)
-			continue
-		if(config.server)	//if you set a server location in config.txt, it sends you there instead of trying to reconnect to the same world address. -- NeoFite
+		if(C && config.server)	//if you set a server location in config.txt, it sends you there instead of trying to reconnect to the same world address. -- NeoFite
 			C << link("byond://[config.server]")
+	if(config.update_check_enabled)
+		install_update()
 
+/world/proc/RoundEndSound(round_end_sound_sent)
```
Comment (ghost) [inline code/world.dm:262]: autistic proc naming, smh

**Pull request merged**